nushell_completions.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2013-2022 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cobra
+
+import (
+	"bytes"
+	"io"
+	"os"
+)
+
+func (c *Command) GenNushellCompletion(w io.Writer) error {
+	buf := new(bytes.Buffer)
+	WriteStringAndCheck(buf, `
+# An external configurator that works with any cobra based
+# command line application (e.g. kubectl, minikube)
+let cobra_configurator = {|spans| 
+ 
+  let cmd = $spans.0
+
+  # skip the first entry in the span (the command) and join the rest of the span to create __complete args
+  let cmd_args = ($spans | skip 1 | str join ' ') 
+
+  # If the last span entry was empty add "" to the end of the command args
+  let cmd_args = if ($spans | last | str trim | is-empty) {
+    $'($cmd_args) ""'
+  } else {
+    $cmd_args
+  }
+
+  # The full command to be executed
+  let full_cmd = $'($cmd) __complete ($cmd_args)'
```

Comment:
I don't know if nushell could support ActiveHelp (see https://github.com/spf13/cobra/blob/main/active_help.md) but unless we do, we should turn it off by setting the environment variable `<PROGRAM>_ACTIVE_HELP=0`.  For example: https://github.com/spf13/cobra/blob/main/fish_completions.go#L55-L56 where the `%[9]s` inserts the name of the variable as specified by the 9th argument to the function

Reply:
Active help isn't supported. Adding the env variable to the latest.